src/components/include/utils/logger.h

```diff
@@ -98,55 +98,86 @@ class Logger {
   static void SetLogger(Pimpl& impl);
   static Pimpl& GetLogger();
 
+  static void SetLoggerPtr(LoggerType& logger) {
+    logger_ = logger;
+  }
+
+  static LoggerType& GetLoggerPtr() {
+    return logger_;
+  }
+
  private:
   Logger();
   Logger(const Logger&);
   Logger& operator=(const Logger&);
   ~Logger();
 
   static Pimpl impl_;
+  static LoggerType logger_;
+};
+
+class Log {
+ public:
+  Log(const bool logs_enabled) {
+#if defined(LOG4CXX_LOGGER)
+
+    logger::Logger::InitLogger(logs_enabled, "log4cxx.properties");
+#else
+    logger::Logger::InitLogger(
+        logs_enabled, logger::LogLevel::LL_TRACE, "SmartDeviceLink.log");
```

Comment:
@pestOO, such kind of logging was done in sdl win. at the moment we're merging and adopting **existing** code. All   not implemented features (or bugs as you say), are out of scope of this merge and should be processed as usual.